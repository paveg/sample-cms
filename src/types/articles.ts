export type Article = {
  id: string;
  title: string;
  content: string;
  data: string;
  createdAt: string;
  updatedAt: string;
  publishedAt: string;
}
