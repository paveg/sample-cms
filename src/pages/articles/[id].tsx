import * as React from 'react';
import { Article } from "../../types/articles";

type Props = {
  article: Article
}

const ArticleId: React.FC<Props> = ({ article }) => {
  return (
    <>
      <h1>{article.title}</h1>
      <p>{article.publishedAt}</p>
      <div>{article.content}</div>
    </>
  )
}

export const getStaticPaths = async () => {
  const res = await fetch('https://samples.microcms.io/api/v1/articles/', {
    headers: {
      'Content-type': 'application/json',
      'X-API-KEY': process.env.MICRO_CMS_API_KEY,
    }
  })
  const json = await res.json()
  const paths = json.contents.map((content) => `/articles/${content.id}`)

  return {
    paths, fallback: false
  }
}

export const getStaticProps = async (context) => {
  const id = context.params.id;
  const res = await fetch('https://samples.microcms.io/api/v1/articles/' + id, {
    headers: {
      'Content-type': 'application/json',
      'X-API-KEY': process.env.MICRO_CMS_API_KEY,
    }
  })
  const json = await res.json()

  return {
    props: {
      article: json
    }
  }
}


export default ArticleId
