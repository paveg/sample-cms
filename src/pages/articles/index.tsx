import * as React from 'react';
import Link from "next/link";
import { Article } from "../../types/articles";

type Props = {
  articles: Article[]
}

const Articles: React.FC<Props> = ({ articles }) => {
  return (
    <div>
      <ul>
        {articles.map((article: Article) => {
          return <li key={article.id}>
            <Link href={`articles/${article.id}`}>
              <a>{article.title}</a>
            </Link>
          </li>
        })}
      </ul>
    </div>
  )
}

export const getStaticProps = async () => {
  const res = await fetch('https://samples.microcms.io/api/v1/articles/', {
    headers: {
      'Content-type': 'application/json',
      'X-API-KEY': process.env.MICRO_CMS_API_KEY,
    }
  })
  const json = await res.json()

  return {
    props: {
      articles: json.contents
    }
  }
}

export default Articles
