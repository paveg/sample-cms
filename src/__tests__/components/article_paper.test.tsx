import * as React from 'react';
import renderer from 'react-test-renderer';
import ArticlePaper from '../../components/article_paper';
import { Article } from '../../types/articles';

const dateString = '2020-11-23T00:00:00.000Z';
const article: Article = {
  id: 'testId',
  title: 'testTitle',
  content: 'testContent',
  date: dateString,
  eyeCatching: { url: 'https://example.com/img' },
  createdAt: dateString,
  updatedAt: dateString,
  publishedAt: dateString,
};

describe('Layout', () => {
  describe('when does not include children', () => {
    const component = renderer.create(<ArticlePaper article={article} />);
    it('renders correctly component', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
