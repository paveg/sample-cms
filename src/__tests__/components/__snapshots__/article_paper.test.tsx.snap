// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout when does not include children renders correctly component 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <span
      className="MuiTypography-root MuiTypography-caption"
    >
      <u>
        2020-10-23
      </u>
    </span>
    <br />
    <img
      className="sc-bdfBwQ dpAPNw"
      src="https://example.com/img"
    />
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        href="/articles/testId"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          testTitle
        </span>
      </a>
    </div>
  </div>
</div>
`;
